{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# we can convert the rules into before, after dictionaries\n",
    "# then also convert each update into a list format\n",
    "# iterate over each element of each update\n",
    "# then check the two sides of the list from the element with the rule dictionaries\n",
    "\n",
    "def parse_input(input_str: str):\n",
    "    rules_str, updates = input_str.split('\\n\\n')\n",
    "    # parse updates into list format\n",
    "    update_list = []\n",
    "    for update_str in updates.split('\\n'):\n",
    "        update = []\n",
    "        for page in update_str.split(','):\n",
    "            update.append(int(page))\n",
    "        update_list.append(update)\n",
    "\n",
    "    # print(update_list)\n",
    "\n",
    "    # parse rules into dictionaries\n",
    "    \n",
    "    # before_dict: keys must come before values\n",
    "    before_dict = dict()\n",
    "    # after_dict: keys come after values\n",
    "    after_dict = dict()\n",
    "\n",
    "    for line in rules_str.split('\\n'):\n",
    "        [before, after] = [int(page) for page in line.split('|')]\n",
    "        if before in before_dict:\n",
    "            before_dict[before].append(after)\n",
    "        else:\n",
    "            before_dict[before] = [after]\n",
    "        if after in after_dict:\n",
    "            after_dict[after].append(before)\n",
    "        else:\n",
    "            after_dict[after] = [before]\n",
    "\n",
    "    # print(before_dict)\n",
    "    # print(after_dict)\n",
    "    \n",
    "    return update_list, before_dict, after_dict\n",
    "\n",
    "class Manual:\n",
    "    def __init__(self, update_list: list, before_rules: dict, after_rules: dict):\n",
    "        self.update_list = update_list\n",
    "        self.before_rules = before_rules\n",
    "        self.after_rules = after_rules\n",
    "\n",
    "    def checkUpdate(self, update: list):\n",
    "        for index, element in enumerate(update):\n",
    "            # splice the list into two parts, before and after element\n",
    "            if index > 0:\n",
    "                pre = update[:index] # list of numbers in the update set that come before the current num \n",
    "                # hence, none of these same numbers should be in the corresponding before_dict value\n",
    "                for item in pre:\n",
    "                    if item in self.before_rules.get(element, []):\n",
    "                        return False\n",
    "        \n",
    "            if index < len(update)-1:\n",
    "                post = update[index+1:]\n",
    "                for item in post:\n",
    "                    if item in self.after_rules.get(element, []):\n",
    "                        return False\n",
    "\n",
    "        return True\n",
    "    \n",
    "    def getMiddleNumber(self, update: list):\n",
    "        # assumption is that list is of odd number length\n",
    "        middle_index = int(len(update) / 2)\n",
    "        return update[middle_index]\n",
    "    \n",
    "    def part1(self):\n",
    "        valid_updates = [update for update in self.update_list if self.checkUpdate(update)]\n",
    "        sum_of_middle_nums = sum([self.getMiddleNumber(update) for update in valid_updates])\n",
    "        return sum_of_middle_nums"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[[75, 47, 61, 53, 29], [97, 61, 53, 29, 13], [75, 29, 13], [75, 97, 47, 61, 53], [61, 13, 29], [97, 13, 75, 29, 47]]\n",
      "{47: [53, 13, 61, 29], 97: [13, 61, 47, 29, 53, 75], 75: [29, 53, 47, 61, 13], 61: [13, 53, 29], 29: [13], 53: [29, 13]}\n",
      "{53: [47, 75, 61, 97], 13: [97, 61, 29, 47, 75, 53], 61: [97, 47, 75], 47: [97, 75], 29: [75, 97, 53, 61, 47], 75: [97]}\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "143"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "with open('data/test/5.txt', 'r', encoding='utf-8') as f:\n",
    "    input_str = f.read()\n",
    "\n",
    "update_list, before_dict, after_dict = parse_input(input_str)\n",
    "manual = Manual(update_list, before_dict, after_dict)\n",
    "manual.part1()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[[52, 77, 83, 31, 94, 75, 34, 95, 29, 38, 82, 19, 41, 39, 27, 98, 84, 13, 55, 21, 66], [76, 17, 43, 93, 46, 67, 68, 28, 18, 32, 87, 15, 89, 27, 31], [28, 63, 89, 27, 13, 66, 98, 95, 19, 36, 55], [38, 21, 39, 86, 16, 62, 76, 85, 47, 35, 93, 67, 68, 28, 18], [46, 67, 68, 28, 18, 87, 15, 89, 27, 31, 52, 13, 75, 66, 29, 98, 95, 19, 44, 77, 84], [41, 34, 98, 29, 95, 96, 55, 86, 52, 13, 16, 38, 82], [55, 83, 21, 16, 62, 76, 56, 43, 46], [37, 14, 56, 17, 18, 16, 35, 67, 21, 38, 59, 85, 93], [63, 18, 55, 27, 98, 32, 84, 19, 89, 87, 52, 44, 31, 95, 77, 36, 66, 13, 28, 29, 15, 94, 75], [55, 76, 39, 16, 82, 19, 34, 66, 29, 96, 75], [77, 34, 84, 94, 36, 55, 83, 82, 41, 38, 21, 39, 96, 86, 16, 62, 76, 56, 85, 59, 47, 35, 37], [62, 17, 47, 39, 37, 86, 85, 74, 38, 14, 93, 16, 67], [37, 67, 56, 46, 85, 62, 43, 68, 96, 17, 86, 39, 14, 76, 41, 82, 16], [89, 44, 67, 98, 28, 13, 75, 31, 63, 87, 52, 84, 34, 68, 77, 32, 46], [41, 38, 39, 96, 62, 17, 56, 47, 35, 37, 93], [44, 84, 94, 36, 21, 16, 62, 76, 85, 47, 35], [34, 13, 32, 83, 18, 89, 94, 31, 84, 27, 29, 95, 19, 75, 55, 63, 44, 52, 66, 98, 77], [75, 68, 67, 89, 74, 15, 28, 63, 44, 98, 29, 93, 52, 18, 95, 43, 13], [84, 83, 41, 16, 62, 17, 85, 47, 35, 14, 43], [66, 75, 29, 39, 36, 82, 19, 13, 95, 44, 84, 96, 38, 16, 34, 52, 55, 86, 98, 83, 21], [38, 39, 37, 93, 17, 59, 68, 82, 85, 46, 47, 16, 67, 86, 62, 14, 35], [75, 34, 27, 98, 89, 28, 84, 63, 36, 32, 87, 55, 31], [43, 93, 74, 46, 67, 68, 28, 87, 63, 15, 89, 31, 52, 13, 75, 66, 29, 98, 95, 19, 44], [28, 87, 63, 15, 31, 52, 13, 29, 95, 19, 34, 84, 94], [32, 87, 63, 15, 89, 31, 13, 75, 66, 95, 19, 77, 34, 94, 36, 83, 82], [55, 21, 17, 56, 83, 98, 76, 62, 39, 38, 19, 29, 44, 41, 84, 34, 95, 86, 16], [17, 56, 59, 35, 14, 93, 74, 46, 67, 27, 52], [89, 27, 52, 19, 77, 34, 94, 21, 39], [85, 14, 74, 46, 68, 28, 18, 32, 87, 63, 15, 27, 52, 13, 75], [83, 82, 41, 38, 21, 39, 96, 86, 16, 62, 76, 17, 56, 85, 59, 47, 35, 37, 14, 43, 93, 74, 67], [85, 14, 74, 46, 68, 32, 87, 15, 31, 52, 75], [95, 27, 19, 77, 98, 31, 84, 94, 15, 63, 34, 75, 55, 83, 52, 87, 13], [13, 66, 19, 77, 34, 84, 55, 86, 62], [34, 84, 87, 19, 13, 83, 31, 29, 44, 95, 15, 77, 55, 36, 52, 27, 66, 98, 75, 89, 41], [16, 62, 76, 56, 47, 37, 43, 93, 87, 63, 89], [15, 13, 75, 66, 34, 84, 36, 83, 82, 38, 21], [14, 27, 46, 18, 75, 59, 63, 47, 85, 67, 68, 37, 43], [63, 15, 89, 27, 31, 13, 66, 98, 95, 19, 38], [18, 32, 63, 15, 27, 31, 52, 13, 75, 98, 95, 19, 44, 77, 34, 84, 36, 55, 83], [44, 75, 27, 63, 98, 18, 34, 66, 68, 31, 19, 95, 52, 32, 28, 87, 29], [19, 44, 77, 34, 84, 94, 36, 55, 83, 82, 41, 38, 21, 96, 86, 16, 62, 76, 17, 56, 85, 59, 47], [67, 63, 18, 93, 37, 46, 17, 87, 89, 43, 52, 56, 35], [98, 18, 29, 89, 31], [52, 13, 66, 98, 95, 19, 34, 84, 94, 55, 82, 41, 39, 86, 16], [67, 28, 87, 75, 66, 29, 19, 77, 34, 84, 94], [31, 13, 55, 41, 98, 38, 15, 83, 27, 95, 29, 89, 84, 77, 21, 34, 75], [18, 31, 13, 75, 95], [27, 14, 87, 63, 18, 32, 29, 67, 31, 89, 15, 37, 13, 43, 74, 47, 68, 28, 35], [93, 96, 82, 41, 94, 16, 17], [17, 56, 85, 37, 43, 93, 74, 67, 68, 28, 18, 32, 63, 15, 89, 31, 52], [96, 16, 76, 17, 85, 59, 47, 37, 46, 68, 18, 87, 63], [17, 16, 77, 59, 95, 44, 34], [77, 44, 75, 68, 32, 29, 67, 74, 15, 52, 46, 87, 13, 31, 34, 63, 28, 66, 95, 98, 27], [68, 63, 15, 13, 66, 19, 84, 94, 36], [13, 75, 66, 29, 98, 95, 19, 44, 77, 34, 84, 94, 36, 55, 83, 41, 38, 21, 39, 96, 86, 16, 62], [15, 27, 52, 13, 29, 19, 44, 77, 94, 36, 55, 83, 41], [43, 93, 67, 68, 28, 18, 32, 87, 63, 15, 89, 27, 52, 13, 75, 66, 98, 19, 44], [55, 83, 38, 39, 86, 16, 76, 14, 93], [29, 84, 87, 27, 34, 94, 18, 55, 13, 63, 77, 98, 89, 66, 95], [41, 21, 96, 59, 35, 46, 28], [83, 82, 41, 38, 21, 39, 96, 86, 62, 76, 17, 56, 85, 59, 47, 35, 14, 43, 67], [82, 94, 39, 76, 98, 75, 86], [14, 43, 74, 46, 68, 18, 32, 63, 15, 89, 13, 75, 66, 98, 95], [94, 39, 82, 41, 75, 55, 98, 89, 27, 66, 19, 34, 21, 84, 95, 77, 29], [85, 59, 35, 37, 43, 93, 74, 46, 67, 68, 28, 18, 32, 87, 15, 89, 27, 31, 52, 13, 75], [59, 47, 35, 37, 14, 43, 93, 74, 46, 67, 68, 28, 18, 32, 87, 63, 15, 89, 27, 31, 52, 13, 66], [16, 76, 17, 56, 47, 35, 14, 93, 74, 46, 68, 18, 63, 15, 89], [39, 96, 16, 62, 76, 56, 85, 59, 47, 35, 14, 93, 67, 28, 18, 32, 87], [56, 85, 59, 47, 35, 37, 14, 43, 93, 74, 46, 67, 68, 28, 18, 32, 87, 63, 15, 89, 27, 52, 13], [36, 82, 38, 39, 62, 37, 74], [66, 59, 75, 63, 15, 35, 74, 28, 67, 93, 37, 27, 18], [66, 84, 41, 36, 87, 44, 31], [15, 77, 34, 44, 67, 32, 66, 29, 18, 68, 13, 28, 31, 27, 19, 63, 89, 95, 84, 94, 98], [68, 35, 86, 46, 14, 93, 85, 37, 96], [27, 31, 52, 75, 66, 29, 98, 95, 19, 77, 84, 36, 55, 82, 41, 38, 96], [98, 38, 85, 55, 76, 84, 44, 83, 95, 36, 96, 82, 86, 34, 17], [55, 89, 66, 29, 63, 77, 87, 75, 84, 31, 94, 19, 34, 36, 52, 32, 95, 98, 27, 18, 44], [66, 29, 98, 95, 19, 44, 77, 94, 36, 38, 96, 86, 16, 76, 17], [16, 46, 85, 17, 59, 74, 18, 39, 47, 37, 67, 93, 76, 56, 32, 96, 14, 28, 62], [34, 86, 59, 83, 56, 94, 76, 84, 17, 55, 36, 77, 35, 39, 38, 96, 16], [34, 84, 36, 98, 95, 39, 44, 66, 75, 76, 55], [41, 56, 21, 82, 86, 16, 47, 59, 19], [59, 47, 35, 37, 14, 93, 74, 46, 67, 68, 28, 18, 32, 87, 63, 15, 89, 27, 31, 52, 13, 75, 66], [41, 39, 38, 13, 82, 34, 83, 98, 31, 95, 55, 19, 86], [29, 52, 84, 31, 98, 95, 77, 63, 94, 83, 15, 66, 38, 19, 41, 55, 89], [86, 16, 62, 56, 85, 47, 35, 37, 14, 46, 68, 87, 15], [35, 93, 37, 89, 15, 46, 29, 13, 18, 75, 68, 14, 66, 87, 52, 27, 67, 74, 31, 43, 32, 28, 47], [13, 82, 94, 75, 55, 19, 63, 66, 77, 34, 98, 41, 31, 95, 89, 44, 87], [41, 38, 21, 39, 96, 86, 16, 62, 76, 17, 56, 85, 59, 47, 35, 37, 14, 43, 74, 46, 67, 68, 28], [14, 43, 93, 74, 46, 67, 68, 28, 18, 32, 87, 63, 15, 89, 27, 52, 13, 75, 66, 29, 98, 95, 19], [21, 39, 86, 76, 17, 56, 59, 37, 43, 46, 28], [67, 32, 63, 93, 37, 31, 29, 18, 74, 15, 66, 47, 52, 43, 13, 75, 89], [56, 85, 59, 47, 35, 37, 14, 43, 93, 74, 46, 68, 18, 32, 87, 63, 15, 89, 27, 31, 52], [87, 18, 75, 27, 63, 28, 19, 67, 29, 43, 74, 95, 31, 98, 52], [85, 38, 56, 84, 98, 86, 95, 34, 62, 36, 16, 44, 77], [37, 67, 87, 63, 27, 52, 75, 98, 95], [37, 62, 21, 86, 84, 83, 82, 14, 47, 55, 59, 34, 96, 35, 36], [96, 86, 16, 62, 76, 56, 85, 59, 47, 35, 37, 14, 43, 74, 46, 67, 28, 18, 32, 87, 63], [75, 66, 29, 19, 94, 55, 83, 82, 38, 21, 16, 62, 76], [35, 43, 74, 75, 66, 29, 98], [85, 47, 35, 17, 28, 87, 32, 59, 76, 37, 39, 56, 93], [18, 14, 68, 67, 63, 62, 27, 32, 89, 43, 46, 28, 15, 93, 87], [77, 34, 94, 55, 83, 82, 41, 38, 21, 39, 86, 16, 76, 17, 59, 35, 37], [74, 21, 18, 43, 14, 96, 32, 62, 67, 56, 59, 76, 85, 39, 46], [34, 94, 36, 82, 41, 38, 76, 17, 85, 35, 14], [66, 86, 21, 36, 84, 82, 44, 34, 55, 83, 52, 13, 94, 75, 29, 98, 16, 19, 39, 77, 96, 38, 41], [16, 34, 86, 76, 59, 41, 39, 94, 21, 47, 83, 84, 62, 38, 82, 36, 96, 17, 56, 55, 35, 77, 85], [86, 62, 76, 56, 85, 59, 47, 35, 37, 43, 93, 74, 67, 32, 15], [35, 14, 43, 93, 74, 46, 67, 68, 28, 18, 32, 87, 63, 15, 89, 27, 31, 52, 13, 75, 66, 29, 98], [85, 27, 59, 47, 67, 35, 37, 15, 62, 56, 46, 89, 76], [37, 14, 43, 93, 74, 46, 67, 68, 28, 32, 63, 31, 52, 13, 75, 66, 29, 98, 95], [77, 27, 83, 98, 32, 31, 18, 63, 19, 66, 94, 55, 29, 36, 34], [77, 84, 94, 55, 83, 38, 76, 17, 56, 59, 47, 35, 37], [44, 87, 32, 13, 98, 28, 94, 66, 52, 89, 77, 67, 75, 95, 34], [59, 35, 37, 14, 93, 46, 18, 32, 87, 15, 89, 27, 31, 52, 13, 75, 66], [41, 47, 28, 74, 17, 68, 43], [93, 74, 46, 67, 68, 28, 18, 32, 87, 15, 89, 27, 31, 52, 13, 75, 66, 29, 98, 95, 19, 44, 77], [85, 59, 35, 68, 28, 18, 89, 31, 75], [32, 87, 15, 89, 44, 84, 94, 36, 83], [84, 34, 29, 41, 94, 66, 75, 27, 38, 63, 52], [98, 19, 77, 34, 84, 36, 55, 41, 21, 39, 96, 86, 16, 62, 76, 56, 85], [27, 52, 75, 66, 19, 44, 77, 34, 94, 55, 82, 38, 39], [62, 76, 17, 85, 59, 47, 35, 37, 14, 43, 93, 74, 67, 68, 28, 63, 27], [82, 39, 16, 62, 17, 59, 47, 14, 46], [89, 27, 52, 13, 75, 66, 98, 95, 19, 44, 77, 34, 84, 94, 36, 55, 41, 38, 21], [17, 74, 46, 15, 56, 43, 37, 32, 14, 68, 76, 31, 47], [63, 38, 27, 31, 44, 98, 55], [21, 41, 39, 59, 86, 82, 36, 94, 55, 38, 47, 93, 17, 14, 85, 43, 16, 76, 37], [75, 98, 95, 44, 77, 55, 38, 16, 76], [39, 96, 16, 62, 76, 56, 59, 37, 43, 93, 74, 67, 28, 18, 87], [86, 77, 59, 94, 76, 83, 44, 96, 39, 35, 85], [55, 83, 96, 59, 93, 74, 46], [86, 17, 93, 67, 28], [29, 98, 19, 44, 77, 34, 84, 94, 36, 55, 83, 39, 86, 16, 62, 76, 56], [31, 52, 75, 66, 95, 77, 34, 84, 36, 83, 21, 39, 86], [46, 67, 68, 85, 15, 18, 17, 59, 63, 28, 56, 87, 31], [89, 27, 31, 52, 75, 29, 19, 44, 34, 36, 82, 38, 21], [31, 52, 67, 19, 87, 29, 44, 77, 63, 28, 94], [62, 16, 41, 14, 17, 86, 35, 59, 21], [44, 86, 21, 34, 59, 94, 85, 96, 76, 35, 41], [21, 95, 75, 15, 19, 77, 66, 94, 36, 13, 52, 82, 98, 44, 29, 31, 84, 38, 83, 27, 41], [38, 39, 96, 16, 62, 56, 85, 59, 35, 37, 14, 74, 46, 67, 68, 28, 18], [55, 21, 96, 86, 37, 14, 46], [98, 63, 27, 44, 15], [39, 96, 86, 62, 76, 56, 85, 35, 37, 74, 46, 67, 68, 18, 87], [83, 59, 37, 38, 82], [98, 19, 34, 83, 82], [39, 19, 59, 84, 47, 77, 16, 34, 56, 44, 83], [56, 47, 83, 62, 55, 44, 21, 59, 35, 76, 39, 16, 86, 17, 82, 77, 41, 36, 38, 96, 34], [14, 28, 18, 32, 27, 31, 29], [83, 21, 89, 52, 39, 34, 41, 77, 31, 44, 29, 55, 84, 38, 98], [68, 31, 13, 98, 44, 84, 36], [34, 84, 82, 86, 76], [29, 32, 19, 63, 66, 44, 89, 52, 46, 98, 77, 18, 75, 67, 84, 13, 95, 27, 68, 15, 28], [63, 89, 27, 31, 75, 29, 98, 19, 84, 94, 36, 55, 83, 82, 41], [35, 86, 67, 28, 37, 76, 93, 63, 68, 59, 18, 47, 32, 96, 17, 43, 16, 14, 85, 62, 56], [29, 52, 55, 19, 13, 15, 89, 83, 31, 77, 75, 63, 82], [77, 34, 36, 55, 83, 82, 41, 96, 16, 62, 56, 85, 37], [35, 59, 86, 37, 74, 47, 62, 46, 38, 76, 14, 83, 39, 43, 82, 41, 55, 16, 17, 93, 96], [56, 59, 47, 37, 74, 18, 32, 89, 27, 52, 13], [31, 93, 89, 63, 35, 15, 52, 87, 74, 68, 32, 98, 67, 37, 28, 18, 46, 14, 75], [74, 47, 15, 43, 63, 35, 86, 62, 67, 56, 93, 68, 46, 16, 32, 87, 17, 37, 18, 59, 14, 28, 76], [93, 74, 63, 89, 31], [47, 35, 37, 14, 43, 93, 74, 46, 67, 68, 28, 32, 87, 63, 15, 89, 27, 31, 52, 13, 75, 66, 29], [68, 63, 15, 31, 98, 95, 19, 44, 77], [93, 46, 68, 28, 18, 87, 63, 15, 89, 31, 52, 13, 66, 29, 95], [77, 85, 95, 21, 94, 39, 41, 86, 98, 17, 34, 56, 83, 84, 82], [63, 87, 14, 46, 43, 47, 56, 13, 67], [62, 76, 17, 43, 93, 74, 68, 63, 27], [85, 47, 43, 93, 74, 68, 63, 15, 89, 27, 52, 13, 75], [17, 35, 37, 14, 93, 46, 68, 87, 15, 27, 52], [27, 34, 77, 98, 19, 32, 75, 89, 13, 18, 28, 74, 68, 46, 95, 52, 44, 67, 29], [95, 13, 66, 27, 84, 44, 36, 89, 63, 55, 28], [41, 38, 21, 39, 96, 86, 16, 62, 76, 56, 85, 59, 47, 35, 37, 14, 43, 93, 46, 67, 28], [52, 13, 75, 66, 29, 98, 95, 19, 44, 77, 34, 84, 94, 55, 83, 82, 41, 38, 21, 39, 96, 86, 16], [86, 56, 35, 59, 83, 37, 82, 55, 96], [89, 47, 63, 18, 74, 62, 28, 43, 59, 27, 14, 76, 67, 15, 93, 68, 85], [86, 87, 67, 43, 47, 39, 14], [18, 55, 89, 77, 19, 13, 66, 29, 28, 32, 52, 27, 94, 95, 44, 15, 36, 87, 34, 31, 84], [35, 37, 14, 43, 93, 74, 46, 67, 68, 28, 18, 32, 63, 15, 89, 27, 31, 52, 13, 75, 66, 29, 98], [77, 34, 84, 36, 41, 38, 39, 96, 86, 76, 56, 85, 59], [31, 66, 84, 94, 82, 21, 86], [93, 32, 87, 52, 75, 66, 29, 19, 77], [74, 46, 27, 18, 37, 28, 14, 87, 93, 66, 35, 67, 52, 15, 29, 75, 98], [98, 66, 29, 75, 15, 68, 27, 67, 31, 13, 63, 93, 89, 46, 52, 87, 95, 18, 32, 19, 44, 74, 43]]\n",
      "{32: [75, 98, 29, 41, 34, 36, 82, 44, 63, 27, 55, 83, 95, 19, 77, 52, 66, 15, 94, 89, 84, 13, 87, 31], 18: [19, 82, 55, 29, 77, 34, 27, 52, 94, 89, 95, 13, 15, 63, 66, 36, 98, 87, 84, 75, 44, 31, 32, 83], 41: [56, 28, 47, 86, 59, 38, 85, 18, 74, 46, 17, 96, 62, 93, 67, 76, 14, 43, 37, 39, 35, 21, 68, 16], 85: [67, 66, 35, 93, 68, 37, 43, 27, 13, 89, 75, 46, 52, 59, 31, 87, 32, 15, 18, 14, 28, 74, 63, 47], 87: [31, 38, 44, 94, 55, 15, 41, 13, 52, 36, 29, 95, 34, 27, 63, 77, 89, 19, 83, 84, 75, 98, 82, 66], 66: [96, 39, 94, 95, 29, 36, 62, 38, 98, 82, 16, 84, 44, 41, 76, 77, 86, 83, 21, 55, 56, 19, 34, 17], 16: [14, 74, 47, 17, 18, 28, 93, 63, 76, 59, 56, 15, 85, 89, 35, 27, 67, 32, 43, 62, 87, 68, 46, 37], 31: [77, 19, 34, 29, 52, 66, 39, 75, 98, 96, 95, 41, 16, 82, 84, 21, 83, 44, 13, 38, 36, 55, 86, 94], 62: [89, 18, 63, 85, 76, 37, 93, 68, 14, 43, 56, 17, 67, 32, 87, 46, 59, 27, 74, 31, 47, 35, 15, 28], 46: [44, 66, 75, 67, 87, 31, 19, 18, 63, 15, 28, 95, 32, 77, 29, 13, 89, 34, 27, 52, 94, 68, 84, 98], 74: [27, 67, 84, 66, 63, 29, 34, 18, 44, 28, 46, 31, 95, 89, 68, 98, 19, 32, 15, 77, 75, 13, 87, 52], 56: [67, 68, 47, 59, 18, 87, 46, 27, 32, 43, 28, 13, 93, 85, 75, 63, 35, 37, 89, 52, 15, 14, 31, 74], 68: [89, 66, 52, 28, 98, 36, 84, 19, 34, 27, 44, 95, 13, 75, 31, 94, 29, 18, 77, 15, 63, 55, 87, 32], 28: [94, 63, 31, 66, 87, 29, 34, 15, 98, 13, 75, 95, 18, 44, 36, 83, 52, 27, 19, 55, 32, 84, 89, 77], 13: [98, 21, 75, 77, 62, 38, 34, 83, 84, 29, 96, 86, 19, 44, 76, 95, 55, 66, 94, 39, 36, 16, 82, 41], 55: [38, 14, 35, 96, 74, 62, 76, 17, 82, 59, 56, 21, 46, 93, 43, 85, 37, 83, 47, 39, 16, 67, 86, 41], 96: [56, 68, 35, 74, 16, 62, 59, 87, 47, 86, 85, 14, 37, 28, 46, 67, 93, 18, 76, 32, 17, 15, 63, 43], 93: [75, 29, 74, 32, 98, 67, 52, 66, 15, 46, 31, 87, 34, 18, 77, 68, 13, 89, 27, 28, 95, 63, 19, 44], 89: [77, 82, 55, 38, 34, 75, 95, 19, 98, 52, 29, 94, 84, 13, 83, 41, 66, 27, 36, 21, 96, 44, 31, 39], 98: [59, 96, 56, 85, 77, 34, 62, 38, 21, 84, 39, 17, 94, 86, 95, 83, 36, 44, 82, 19, 16, 55, 76, 41], 37: [28, 87, 14, 74, 75, 29, 63, 31, 32, 93, 46, 15, 27, 95, 18, 68, 67, 13, 43, 98, 52, 19, 66, 89], 34: [35, 41, 43, 85, 94, 84, 59, 56, 37, 96, 86, 14, 38, 17, 82, 21, 76, 47, 39, 55, 83, 16, 36, 62], 38: [47, 96, 17, 46, 32, 16, 39, 37, 56, 18, 86, 76, 43, 68, 21, 93, 59, 67, 74, 85, 35, 14, 28, 62], 35: [18, 28, 29, 89, 63, 74, 43, 93, 98, 52, 37, 87, 14, 68, 46, 27, 13, 67, 95, 66, 15, 31, 75, 32], 44: [94, 17, 76, 41, 16, 85, 36, 38, 47, 59, 37, 39, 77, 35, 86, 55, 83, 84, 96, 21, 34, 82, 62, 56], 67: [77, 75, 94, 19, 32, 15, 66, 36, 18, 31, 52, 44, 27, 84, 89, 28, 13, 29, 63, 34, 98, 87, 68, 95], 29: [34, 21, 16, 56, 41, 84, 38, 76, 17, 62, 82, 19, 96, 36, 83, 55, 98, 85, 94, 95, 86, 77, 39, 44], 14: [13, 29, 74, 31, 75, 18, 89, 15, 52, 67, 98, 66, 95, 93, 87, 28, 27, 32, 44, 68, 19, 63, 43, 46], 59: [28, 29, 46, 52, 68, 15, 32, 18, 87, 37, 67, 13, 47, 63, 93, 75, 66, 35, 43, 31, 89, 74, 27, 14], 82: [85, 14, 93, 56, 37, 59, 62, 35, 28, 43, 41, 76, 47, 96, 68, 46, 74, 16, 17, 67, 86, 21, 38, 39], 17: [89, 15, 13, 59, 68, 43, 63, 52, 28, 74, 32, 27, 14, 18, 67, 37, 31, 93, 87, 47, 85, 56, 46, 35], 19: [41, 38, 84, 39, 44, 76, 83, 94, 96, 16, 17, 82, 34, 21, 36, 55, 62, 47, 35, 59, 56, 77, 86, 85], 77: [34, 96, 94, 62, 16, 47, 84, 85, 35, 86, 17, 36, 56, 83, 14, 76, 37, 38, 55, 21, 41, 82, 59, 39], 43: [87, 46, 27, 93, 18, 66, 52, 77, 44, 19, 13, 29, 15, 89, 31, 67, 74, 63, 75, 95, 32, 28, 98, 68], 15: [95, 31, 84, 19, 82, 27, 21, 38, 98, 75, 52, 55, 66, 13, 34, 29, 39, 94, 77, 36, 89, 41, 83, 44], 76: [46, 17, 27, 43, 89, 74, 35, 28, 68, 67, 56, 52, 15, 87, 85, 59, 37, 31, 18, 47, 32, 93, 63, 14], 95: [59, 82, 47, 17, 34, 94, 77, 55, 86, 84, 21, 96, 39, 83, 44, 56, 41, 36, 19, 38, 76, 62, 16, 85], 83: [76, 96, 14, 56, 68, 16, 93, 46, 17, 37, 59, 86, 62, 82, 74, 85, 21, 35, 47, 38, 43, 67, 39, 41], 75: [34, 84, 82, 36, 98, 19, 83, 17, 44, 55, 76, 39, 77, 66, 62, 21, 94, 86, 41, 96, 29, 16, 95, 38], 52: [29, 66, 83, 19, 84, 95, 13, 98, 16, 77, 75, 41, 96, 39, 21, 82, 36, 86, 94, 38, 62, 44, 55, 34], 21: [59, 76, 18, 86, 39, 46, 37, 16, 62, 96, 14, 35, 56, 68, 43, 67, 32, 47, 17, 85, 74, 93, 87, 28], 84: [83, 85, 41, 14, 62, 59, 16, 93, 39, 38, 36, 55, 43, 86, 17, 47, 82, 21, 56, 94, 96, 76, 35, 37], 47: [68, 35, 74, 67, 89, 31, 63, 27, 52, 87, 15, 28, 14, 66, 46, 32, 29, 37, 98, 43, 13, 75, 93, 18], 39: [43, 56, 32, 14, 96, 85, 93, 74, 28, 18, 17, 86, 47, 16, 46, 59, 62, 35, 63, 67, 68, 87, 76, 37], 27: [29, 21, 77, 94, 66, 39, 82, 86, 13, 31, 75, 41, 19, 98, 34, 36, 84, 44, 55, 83, 96, 52, 95, 38], 36: [74, 59, 55, 86, 43, 93, 38, 16, 85, 37, 96, 39, 56, 14, 17, 46, 83, 82, 21, 35, 47, 41, 62, 76], 94: [56, 86, 85, 17, 37, 36, 59, 74, 93, 39, 38, 76, 55, 82, 62, 21, 83, 47, 43, 41, 16, 35, 96, 14], 86: [85, 74, 47, 32, 16, 28, 35, 46, 14, 56, 15, 76, 63, 93, 59, 89, 87, 17, 37, 43, 67, 18, 68, 62], 63: [15, 77, 38, 44, 19, 27, 83, 41, 21, 82, 34, 94, 95, 36, 89, 31, 52, 84, 13, 75, 98, 29, 55, 66]}\n",
      "{75: [32, 31, 46, 28, 13, 93, 89, 37, 35, 67, 14, 59, 43, 15, 52, 47, 27, 63, 18, 85, 87, 74, 56, 68], 19: [18, 31, 46, 68, 13, 89, 98, 67, 29, 14, 43, 15, 95, 75, 52, 27, 63, 32, 87, 66, 74, 28, 93, 37], 82: [18, 55, 89, 98, 34, 44, 29, 19, 77, 15, 95, 83, 75, 52, 84, 27, 36, 94, 63, 32, 87, 66, 31, 13], 56: [41, 55, 96, 98, 34, 38, 44, 29, 82, 17, 19, 77, 76, 95, 83, 21, 84, 39, 36, 94, 86, 66, 16, 62], 28: [41, 16, 74, 56, 68, 96, 37, 38, 35, 67, 14, 59, 82, 17, 43, 76, 21, 47, 39, 86, 85, 62, 46, 93], 47: [41, 16, 56, 96, 34, 38, 44, 59, 82, 17, 19, 77, 76, 95, 83, 21, 84, 39, 36, 94, 86, 85, 62, 55], 67: [85, 46, 74, 56, 96, 93, 37, 38, 35, 14, 59, 82, 17, 43, 76, 83, 21, 47, 39, 86, 41, 16, 62, 55], 66: [85, 31, 46, 74, 68, 28, 93, 89, 35, 67, 14, 59, 43, 15, 75, 52, 47, 27, 63, 32, 18, 87, 13, 37], 35: [85, 55, 96, 34, 38, 44, 59, 82, 17, 19, 77, 76, 83, 21, 84, 47, 39, 36, 94, 86, 41, 16, 62, 56], 93: [85, 16, 62, 55, 96, 37, 38, 35, 14, 59, 82, 17, 43, 76, 83, 21, 84, 47, 39, 36, 94, 86, 41, 56], 31: [87, 46, 28, 93, 37, 35, 67, 14, 59, 17, 43, 15, 76, 47, 27, 63, 32, 18, 85, 62, 74, 56, 68, 89], 38: [87, 13, 55, 89, 98, 34, 44, 29, 82, 19, 77, 15, 95, 83, 75, 52, 84, 27, 36, 94, 63, 41, 66, 31], 44: [87, 46, 74, 68, 28, 13, 98, 67, 29, 14, 19, 43, 15, 95, 75, 52, 27, 63, 32, 18, 66, 31, 93, 89], 94: [87, 66, 28, 89, 98, 34, 44, 67, 29, 19, 77, 15, 95, 75, 52, 84, 27, 63, 32, 18, 31, 46, 68, 13], 55: [87, 89, 34, 44, 29, 19, 77, 15, 95, 75, 52, 84, 27, 36, 94, 63, 32, 18, 66, 31, 68, 28, 13, 98], 96: [66, 13, 55, 98, 34, 38, 44, 29, 82, 19, 77, 95, 83, 75, 52, 21, 84, 39, 27, 36, 94, 41, 31, 89], 39: [66, 31, 98, 34, 38, 44, 29, 82, 19, 77, 15, 95, 83, 75, 52, 21, 84, 27, 36, 94, 41, 13, 55, 89], 95: [66, 68, 28, 89, 98, 37, 35, 67, 29, 14, 43, 15, 75, 52, 27, 63, 32, 18, 87, 31, 46, 74, 13, 93], 29: [66, 31, 74, 28, 13, 93, 89, 37, 35, 67, 14, 59, 43, 15, 75, 52, 47, 27, 63, 32, 18, 87, 46, 68], 36: [66, 68, 89, 98, 44, 67, 29, 19, 77, 15, 95, 75, 52, 84, 27, 94, 63, 32, 18, 87, 31, 28, 13, 34], 14: [16, 62, 55, 96, 37, 34, 38, 35, 59, 82, 17, 77, 76, 83, 21, 84, 47, 39, 36, 94, 86, 41, 85, 56], 74: [16, 55, 96, 93, 37, 38, 35, 14, 59, 82, 17, 43, 76, 83, 21, 47, 39, 36, 94, 86, 41, 85, 62, 56], 17: [16, 55, 98, 34, 38, 44, 29, 82, 19, 77, 76, 95, 83, 75, 21, 84, 39, 36, 94, 86, 41, 66, 62, 96], 18: [16, 62, 46, 74, 56, 28, 93, 37, 38, 35, 67, 14, 59, 17, 43, 76, 21, 47, 39, 86, 41, 85, 68, 96], 77: [31, 13, 93, 89, 98, 44, 67, 29, 19, 43, 15, 95, 75, 52, 27, 63, 32, 18, 87, 66, 46, 74, 68, 28], 34: [31, 74, 68, 28, 13, 93, 89, 98, 44, 67, 29, 19, 77, 15, 95, 75, 52, 27, 63, 32, 18, 87, 66, 46], 52: [31, 68, 93, 89, 37, 35, 67, 14, 59, 17, 43, 15, 76, 47, 27, 63, 32, 18, 85, 87, 46, 74, 56, 28], 89: [62, 68, 93, 35, 67, 14, 59, 17, 43, 15, 76, 47, 86, 63, 32, 18, 85, 87, 16, 46, 74, 56, 28, 37], 63: [62, 46, 74, 28, 37, 35, 67, 14, 59, 17, 43, 76, 47, 39, 86, 32, 18, 85, 87, 16, 56, 68, 96, 93], 85: [62, 55, 96, 98, 34, 38, 44, 29, 82, 17, 19, 77, 76, 95, 83, 21, 84, 39, 36, 94, 86, 41, 16, 56], 76: [62, 13, 55, 34, 38, 44, 29, 82, 19, 77, 95, 83, 75, 21, 84, 39, 36, 94, 86, 41, 66, 16, 96, 98], 37: [62, 96, 34, 38, 35, 44, 59, 82, 17, 77, 76, 83, 21, 84, 47, 39, 36, 94, 86, 41, 85, 16, 56, 55], 68: [62, 56, 96, 93, 37, 38, 35, 67, 14, 59, 82, 17, 43, 76, 83, 21, 47, 39, 86, 41, 85, 16, 46, 74], 87: [46, 56, 28, 96, 93, 37, 35, 67, 14, 59, 17, 43, 76, 21, 47, 39, 86, 32, 18, 85, 16, 62, 74, 68], 15: [46, 28, 93, 37, 35, 67, 14, 59, 17, 43, 76, 47, 86, 63, 32, 18, 85, 87, 16, 62, 74, 56, 68, 96], 27: [74, 56, 68, 89, 37, 35, 67, 14, 59, 17, 43, 15, 76, 47, 63, 32, 18, 85, 87, 16, 62, 46, 28, 93], 84: [74, 68, 13, 89, 98, 34, 44, 67, 29, 19, 77, 15, 95, 75, 52, 27, 63, 32, 18, 87, 66, 31, 46, 28], 46: [74, 56, 55, 96, 93, 37, 38, 35, 14, 59, 82, 17, 43, 76, 83, 21, 47, 39, 36, 86, 41, 85, 16, 62], 59: [56, 55, 96, 98, 34, 38, 44, 82, 17, 19, 77, 76, 95, 83, 21, 84, 39, 36, 94, 86, 41, 85, 16, 62], 32: [56, 93, 37, 38, 35, 67, 14, 59, 17, 43, 76, 21, 47, 39, 86, 18, 85, 16, 62, 46, 74, 68, 28, 96], 43: [56, 55, 37, 34, 38, 35, 14, 59, 82, 17, 76, 83, 21, 84, 47, 39, 36, 94, 86, 41, 85, 16, 62, 96], 13: [56, 68, 28, 93, 89, 37, 35, 67, 14, 59, 17, 43, 15, 52, 47, 27, 63, 32, 18, 85, 87, 31, 46, 74], 98: [68, 28, 13, 93, 89, 37, 35, 67, 29, 14, 43, 15, 75, 52, 47, 27, 63, 32, 18, 87, 66, 31, 46, 74], 21: [13, 55, 98, 34, 38, 44, 29, 82, 19, 77, 15, 95, 83, 75, 52, 84, 27, 36, 94, 63, 41, 66, 31, 89], 62: [13, 55, 96, 98, 44, 29, 82, 19, 77, 95, 83, 75, 52, 21, 84, 39, 36, 94, 86, 41, 66, 16, 34, 38], 83: [13, 89, 98, 34, 44, 29, 19, 77, 15, 95, 75, 52, 84, 27, 36, 94, 63, 32, 18, 87, 66, 31, 28, 55], 86: [13, 96, 98, 34, 38, 44, 29, 82, 19, 77, 95, 83, 75, 52, 21, 84, 39, 27, 36, 94, 41, 66, 31, 55], 16: [96, 34, 38, 44, 29, 82, 19, 77, 95, 83, 75, 52, 21, 84, 39, 36, 94, 86, 41, 66, 31, 13, 55, 98], 41: [89, 34, 44, 29, 82, 19, 77, 15, 95, 83, 75, 52, 84, 27, 36, 94, 63, 32, 87, 66, 31, 13, 55, 98]}\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "5091"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "with open('data/input/5.txt', 'r', encoding='utf-8') as f:\n",
    "    input_str = f.read()\n",
    "\n",
    "update_list, before_dict, after_dict = parse_input(input_str)\n",
    "manual = Manual(update_list, before_dict, after_dict)\n",
    "manual.part1()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "# part 2: get only incorrectly ordered updates and fix them \n",
    "# need to fix the entries\n",
    "# and then get the middle number of the fixed entry\n",
    "\n",
    "# to fix entries, we just need to check the after rules and iterate from left to right\n",
    "# we can use two stacks to 'sort' the numbers\n",
    "# add all the elements of the list to the second stack \n",
    "# push the first element of the second stack onto the first stack\n",
    "# from the second element onwards, withhold the number and check if prior numbers in the first stack are ok\n",
    "# push numbers from the first stack back to second stack until sequence is ok, then push the withheld number into the first stack\n",
    "# continue this sequence until the second stack is empty\n",
    "\n",
    "from collections import deque\n",
    "from tqdm.notebook import tqdm\n",
    "\n",
    "class Manual2(Manual):\n",
    "    def fixUpdate(self, update: list):\n",
    "        stack1 = deque()\n",
    "        stack2 = deque()\n",
    "        # add all elements of list to second stack\n",
    "        for item in update:\n",
    "            stack2.append(item)\n",
    "\n",
    "        # push the first element of second stack to the first stack\n",
    "        # this first element is now assumed to be the start of the list (for now)\n",
    "        item = stack2.pop()\n",
    "        stack1.append(item)\n",
    "\n",
    "        while stack2:\n",
    "            item = stack2.pop()\n",
    "            # if this popped item must come before some items in stack1, we need to pop everything in stack1 until we can safely add the item to stack1\n",
    "            # check the before dict to see if there are any items that must come after the popped item\n",
    "            constraints = self.after_rules.get(item, [])\n",
    "            for c in constraints:\n",
    "                if stack1:\n",
    "                    if c in stack1:\n",
    "                        stack2.append(stack1.pop())\n",
    "                else:\n",
    "                    break\n",
    "            # now we can safely push the withheld item to stack1\n",
    "            stack1.append(item)\n",
    "\n",
    "        return list(stack1)[::-1]\n",
    "        \n",
    "\n",
    "    def part2(self):\n",
    "        invalid_updates = [update for update in self.update_list if not self.checkUpdate(update)]\n",
    "        print(len(invalid_updates))\n",
    "        fixed_updates = tqdm([self.fixUpdate(update) for update in invalid_updates])\n",
    "        sum_of_middle_nums = sum([self.getMiddleNumber(update) for update in fixed_updates])\n",
    "        return sum_of_middle_nums\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "f2c3c340722842c5b2dcc7a34bbb480e",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "  0%|          | 0/3 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "123"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "with open('data/test/5.txt', 'r', encoding='utf-8') as f:\n",
    "    input_str = f.read()\n",
    "\n",
    "update_list, before_dict, after_dict = parse_input(input_str)\n",
    "manual = Manual2(update_list, before_dict, after_dict)\n",
    "manual.part2()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "86\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  0%|          | 0/86 [00:00<?, ?it/s]Process SpawnPoolWorker-10:\n",
      "Process SpawnPoolWorker-8:\n",
      "Process SpawnPoolWorker-9:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-11:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-12:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-13:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-14:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-15:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-16:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-17:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-18:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-19:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-20:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-21:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-23:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-22:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-24:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-25:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-26:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-27:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-28:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-29:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-30:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-31:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-32:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-34:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-33:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-35:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-36:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-37:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-38:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-39:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-40:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-41:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-42:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-43:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-44:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-45:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-46:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-47:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-48:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-49:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-50:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-51:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-52:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-53:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-54:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-55:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-56:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-57:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-58:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-59:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-61:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-60:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-62:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-63:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-64:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-65:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-66:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-67:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-68:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-69:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-70:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-71:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-72:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-73:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-74:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-75:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-76:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-77:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-78:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-79:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-80:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-81:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-82:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-83:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-84:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-86:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-85:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-87:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-88:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-89:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-90:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-91:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-92:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-93:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "  0%|          | 0/86 [00:03<?, ?it/s]\n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mIndexError\u001b[0m                                Traceback (most recent call last)",
      "File \u001b[0;32m~/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py:856\u001b[0m, in \u001b[0;36mIMapIterator.next\u001b[0;34m(self, timeout)\u001b[0m\n\u001b[1;32m    855\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m--> 856\u001b[0m     item \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_items\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mpopleft\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    857\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mIndexError\u001b[39;00m:\n",
      "\u001b[0;31mIndexError\u001b[0m: pop from an empty deque",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[33], line 8\u001b[0m\n\u001b[1;32m      6\u001b[0m update_list, before_dict, after_dict \u001b[38;5;241m=\u001b[39m parse_input(input_str)\n\u001b[1;32m      7\u001b[0m manual \u001b[38;5;241m=\u001b[39m Manual2(update_list, before_dict, after_dict)\n\u001b[0;32m----> 8\u001b[0m \u001b[43mmanual\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mpart2\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n",
      "Cell \u001b[0;32mIn[31], line 32\u001b[0m, in \u001b[0;36mManual2.part2\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m     30\u001b[0m num_processes \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mmin\u001b[39m(cpu_count(), \u001b[38;5;241m4\u001b[39m)  \u001b[38;5;66;03m# Use up to 4 processes\u001b[39;00m\n\u001b[1;32m     31\u001b[0m \u001b[38;5;28;01mwith\u001b[39;00m Pool(num_processes) \u001b[38;5;28;01mas\u001b[39;00m pool:\n\u001b[0;32m---> 32\u001b[0m     fixed_updates \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mlist\u001b[39;49m\u001b[43m(\u001b[49m\u001b[43mtqdm\u001b[49m\u001b[43m(\u001b[49m\u001b[43mpool\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mimap\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mfixUpdate\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43minvalid_updates\u001b[49m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mtotal\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mlen\u001b[39;49m\u001b[43m(\u001b[49m\u001b[43minvalid_updates\u001b[49m\u001b[43m)\u001b[49m\u001b[43m)\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m     34\u001b[0m sum_of_middle_nums \u001b[38;5;241m=\u001b[39m \u001b[38;5;28msum\u001b[39m([\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mgetMiddleNumber(update) \u001b[38;5;28;01mfor\u001b[39;00m update \u001b[38;5;129;01min\u001b[39;00m fixed_updates])\n\u001b[1;32m     35\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m sum_of_middle_nums\n",
      "File \u001b[0;32m~/anaconda3/envs/aoc_24/lib/python3.10/site-packages/tqdm/std.py:1181\u001b[0m, in \u001b[0;36mtqdm.__iter__\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m   1178\u001b[0m time \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_time\n\u001b[1;32m   1180\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m-> 1181\u001b[0m     \u001b[38;5;28;01mfor\u001b[39;00m obj \u001b[38;5;129;01min\u001b[39;00m iterable:\n\u001b[1;32m   1182\u001b[0m         \u001b[38;5;28;01myield\u001b[39;00m obj\n\u001b[1;32m   1183\u001b[0m         \u001b[38;5;66;03m# Update and possibly print the progressbar.\u001b[39;00m\n\u001b[1;32m   1184\u001b[0m         \u001b[38;5;66;03m# Note: does not call self.update(1) for speed optimisation.\u001b[39;00m\n",
      "File \u001b[0;32m~/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py:861\u001b[0m, in \u001b[0;36mIMapIterator.next\u001b[0;34m(self, timeout)\u001b[0m\n\u001b[1;32m    859\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_pool \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m\n\u001b[1;32m    860\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mStopIteration\u001b[39;00m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m\n\u001b[0;32m--> 861\u001b[0m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_cond\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mwait\u001b[49m\u001b[43m(\u001b[49m\u001b[43mtimeout\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    862\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[1;32m    863\u001b[0m     item \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_items\u001b[38;5;241m.\u001b[39mpopleft()\n",
      "File \u001b[0;32m~/anaconda3/envs/aoc_24/lib/python3.10/threading.py:320\u001b[0m, in \u001b[0;36mCondition.wait\u001b[0;34m(self, timeout)\u001b[0m\n\u001b[1;32m    318\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:    \u001b[38;5;66;03m# restore state no matter what (e.g., KeyboardInterrupt)\u001b[39;00m\n\u001b[1;32m    319\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m timeout \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[0;32m--> 320\u001b[0m         \u001b[43mwaiter\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43macquire\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    321\u001b[0m         gotit \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mTrue\u001b[39;00m\n\u001b[1;32m    322\u001b[0m     \u001b[38;5;28;01melse\u001b[39;00m:\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "with open('data/input/5.txt', 'r', encoding='utf-8') as f:\n",
    "    input_str = f.read()\n",
    "\n",
    "update_list, before_dict, after_dict = parse_input(input_str)\n",
    "manual = Manual2(update_list, before_dict, after_dict)\n",
    "manual.part2()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "86\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  0%|          | 0/86 [00:00<?, ?it/s]Process SpawnPoolWorker-101:\n",
      "Process SpawnPoolWorker-98:\n",
      "Process SpawnPoolWorker-100:\n",
      "Process SpawnPoolWorker-99:\n",
      "Traceback (most recent call last):\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "Traceback (most recent call last):\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-102:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-104:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-103:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-105:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-106:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-107:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-108:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-109:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-110:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-111:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-112:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-113:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-114:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-115:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-116:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-117:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-118:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-119:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-120:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-121:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-122:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-123:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-124:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-125:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-126:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-127:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-128:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-129:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-130:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-131:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-132:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-133:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-134:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-136:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-135:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-137:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-138:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-139:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-140:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-141:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-142:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-144:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-143:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-145:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-146:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-147:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-148:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-149:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-150:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-151:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-152:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-153:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-154:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-155:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-156:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-157:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-158:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-159:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-160:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-161:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-162:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-163:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-164:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-165:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-166:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-167:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-168:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-169:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-170:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-171:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-172:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-173:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-174:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-175:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-176:\n",
      "Process SpawnPoolWorker-177:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-178:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-179:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-180:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-181:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-182:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "Process SpawnPoolWorker-183:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 314, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/process.py\", line 108, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py\", line 114, in worker\n",
      "    task = get()\n",
      "  File \"/Users/kieron/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/queues.py\", line 367, in get\n",
      "    return _ForkingPickler.loads(res)\n",
      "AttributeError: Can't get attribute 'Manual2' on <module '__main__' (built-in)>\n",
      "  0%|          | 0/86 [00:53<?, ?it/s]\n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mIndexError\u001b[0m                                Traceback (most recent call last)",
      "File \u001b[0;32m~/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py:856\u001b[0m, in \u001b[0;36mIMapIterator.next\u001b[0;34m(self, timeout)\u001b[0m\n\u001b[1;32m    855\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m--> 856\u001b[0m     item \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_items\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mpopleft\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    857\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mIndexError\u001b[39;00m:\n",
      "\u001b[0;31mIndexError\u001b[0m: pop from an empty deque",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[36], line 44\u001b[0m\n\u001b[1;32m     42\u001b[0m update_list, before_dict, after_dict \u001b[38;5;241m=\u001b[39m parse_input(input_str)\n\u001b[1;32m     43\u001b[0m manual \u001b[38;5;241m=\u001b[39m Manual2(update_list, before_dict, after_dict)\n\u001b[0;32m---> 44\u001b[0m \u001b[43mmanual\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mpart2\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n",
      "Cell \u001b[0;32mIn[36], line 32\u001b[0m, in \u001b[0;36mManual2.part2\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m     30\u001b[0m num_processes \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mmin\u001b[39m(cpu_count(), \u001b[38;5;241m4\u001b[39m)  \u001b[38;5;66;03m# Use up to 4 processes\u001b[39;00m\n\u001b[1;32m     31\u001b[0m \u001b[38;5;28;01mwith\u001b[39;00m Pool(num_processes) \u001b[38;5;28;01mas\u001b[39;00m pool:\n\u001b[0;32m---> 32\u001b[0m     fixed_updates \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mlist\u001b[39;49m\u001b[43m(\u001b[49m\u001b[43mtqdm\u001b[49m\u001b[43m(\u001b[49m\u001b[43mpool\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mimap\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mfixUpdate\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43minvalid_updates\u001b[49m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mtotal\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mlen\u001b[39;49m\u001b[43m(\u001b[49m\u001b[43minvalid_updates\u001b[49m\u001b[43m)\u001b[49m\u001b[43m)\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m     34\u001b[0m sum_of_middle_nums \u001b[38;5;241m=\u001b[39m \u001b[38;5;28msum\u001b[39m([\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mgetMiddleNumber(update) \u001b[38;5;28;01mfor\u001b[39;00m update \u001b[38;5;129;01min\u001b[39;00m fixed_updates])\n\u001b[1;32m     35\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m sum_of_middle_nums\n",
      "File \u001b[0;32m~/anaconda3/envs/aoc_24/lib/python3.10/site-packages/tqdm/std.py:1181\u001b[0m, in \u001b[0;36mtqdm.__iter__\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m   1178\u001b[0m time \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_time\n\u001b[1;32m   1180\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m-> 1181\u001b[0m     \u001b[38;5;28;01mfor\u001b[39;00m obj \u001b[38;5;129;01min\u001b[39;00m iterable:\n\u001b[1;32m   1182\u001b[0m         \u001b[38;5;28;01myield\u001b[39;00m obj\n\u001b[1;32m   1183\u001b[0m         \u001b[38;5;66;03m# Update and possibly print the progressbar.\u001b[39;00m\n\u001b[1;32m   1184\u001b[0m         \u001b[38;5;66;03m# Note: does not call self.update(1) for speed optimisation.\u001b[39;00m\n",
      "File \u001b[0;32m~/anaconda3/envs/aoc_24/lib/python3.10/multiprocessing/pool.py:861\u001b[0m, in \u001b[0;36mIMapIterator.next\u001b[0;34m(self, timeout)\u001b[0m\n\u001b[1;32m    859\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_pool \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m\n\u001b[1;32m    860\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mStopIteration\u001b[39;00m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m\n\u001b[0;32m--> 861\u001b[0m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_cond\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mwait\u001b[49m\u001b[43m(\u001b[49m\u001b[43mtimeout\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    862\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[1;32m    863\u001b[0m     item \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_items\u001b[38;5;241m.\u001b[39mpopleft()\n",
      "File \u001b[0;32m~/anaconda3/envs/aoc_24/lib/python3.10/threading.py:320\u001b[0m, in \u001b[0;36mCondition.wait\u001b[0;34m(self, timeout)\u001b[0m\n\u001b[1;32m    318\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:    \u001b[38;5;66;03m# restore state no matter what (e.g., KeyboardInterrupt)\u001b[39;00m\n\u001b[1;32m    319\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m timeout \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[0;32m--> 320\u001b[0m         \u001b[43mwaiter\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43macquire\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    321\u001b[0m         gotit \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mTrue\u001b[39;00m\n\u001b[1;32m    322\u001b[0m     \u001b[38;5;28;01melse\u001b[39;00m:\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "from multiprocess import Pool, cpu_count\n",
    "from tqdm import tqdm\n",
    "\n",
    "class Manual2(Manual):\n",
    "    def fixUpdate(self, update: list):\n",
    "        # Your existing `fixUpdate` implementation\n",
    "        stack1 = deque()\n",
    "        stack2 = deque()\n",
    "        for item in update:\n",
    "            stack2.append(item)\n",
    "        item = stack2.pop()\n",
    "        stack1.append(item)\n",
    "        while stack2:\n",
    "            item = stack2.pop()\n",
    "            constraints = self.after_rules.get(item, [])\n",
    "            for c in constraints:\n",
    "                if stack1:\n",
    "                    if c in stack1:\n",
    "                        stack2.append(stack1.pop())\n",
    "                else:\n",
    "                    break\n",
    "            stack1.append(item)\n",
    "        return list(stack1)[::-1]\n",
    "\n",
    "    def part2(self):\n",
    "        invalid_updates = [update for update in self.update_list if not self.checkUpdate(update)]\n",
    "        print(len(invalid_updates))\n",
    "\n",
    "        # Limit the number of processes to avoid overloading\n",
    "        num_processes = min(cpu_count(), 4)  # Use up to 4 processes\n",
    "        with Pool(num_processes) as pool:\n",
    "            fixed_updates = list(tqdm(pool.imap(self.fixUpdate, invalid_updates), total=len(invalid_updates)))\n",
    "\n",
    "        sum_of_middle_nums = sum([self.getMiddleNumber(update) for update in fixed_updates])\n",
    "        return sum_of_middle_nums\n",
    "\n",
    "if __name__ == '__main__':\n",
    "    # Initialize the manual object and call `part2`\n",
    "    with open('data/input/5.txt', 'r', encoding='utf-8') as f:\n",
    "        input_str = f.read()\n",
    "\n",
    "    update_list, before_dict, after_dict = parse_input(input_str)\n",
    "    manual = Manual2(update_list, before_dict, after_dict)\n",
    "    manual.part2()\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "aoc_24",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
